sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -300,34 +341,54 @@ public String subclass(TypeDescription.Generic superClass) {
                     return super.name(clazzDescription);
                   }
                 })
-            // Create a subclass of DoFnInvoker
-            .subclass(DoFnInvoker.class, ConstructorStrategy.Default.NO_CONSTRUCTORS)
-            .defineField(
-                FN_DELEGATE_FIELD_NAME, fnClass, Visibility.PRIVATE, FieldManifestation.FINAL)
+            // class <invoker class> extends DoFnInvokerBase {
+            .subclass(DoFnInvokerBase.class, ConstructorStrategy.Default.NO_CONSTRUCTORS)
+
+            //   public <invoker class>(<fn class> delegate) { this.delegate = delegate; }
```

Comment:
I tried and failed to delegate the constructor rather than generating a `StackManipulation`. It _should_ be as easy as delegating to something like the following (composed with a `SuperCall`)

```java
   void construct(@This DoFnInvokerBase thiz, @Argument(0) DoFn delegate) { thiz.delegate = delegate; }
```

but I failed to find the appropriate incantation. Since it is a "solved" problem for now, I am leaving it alone.